src/OrchardCore.Modules/OrchardCore.OpenApi.Swagger/Startup.cs

```diff
@@ -0,0 +1,69 @@
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.ApiExplorer;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Modules;
+using Swashbuckle.AspNetCore.SwaggerGen;
+using System;
+using System.Collections.Generic;
+
+namespace OrchardCore.OpenApi.Swagger
+{
+    public class Startup : StartupBase
+    {
+        public override void Configure(IApplicationBuilder app, IEndpointRouteBuilder routes, IServiceProvider serviceProvider)
+        {
+            var httpContextAccessor = serviceProvider.GetRequiredService<IHttpContextAccessor>();
+
+            app.UseSwagger();
+            app.UseSwaggerUI(c =>
+            {
+                c.RoutePrefix = "swagger";
+
+                var swaggerPath = httpContextAccessor.HttpContext.Request.PathBase + new PathString("/swagger");
+
+                foreach (var definition in serviceProvider.GetServices<IOpenApiDefinition>())
+                {
+                    c.SwaggerEndpoint($"{swaggerPath}/{definition.Name}/swagger.json", definition.Name);
```

Comment:
Is swagger.json a standard swagger convention?

Reply:
@Jetski5822 I think so

Reply:
cool